src/adapters/methods/adapter_layer_base.py

```diff
@@ -133,16 +146,38 @@ def enable_adapters(self, adapter_setup: AdapterCompositionBlock, unfreeze_adapt
             unfreeze_adapters (bool): Whether to unfreeze the adapters.
             unfreeze_fusion (bool): Whether to unfreeze the fusion layers.
         """
-        raise NotImplementedError()
+        if unfreeze_adapters:
+            for name in adapter_setup.flatten():
+                if name in self.adapter_modules:
+                    for param in self.adapter_modules[name].parameters():
+                        param.requires_grad = True
+
+    def freeze_adapter(self, adapter_name: str, freeze: bool = True):
+        """Freezes/ unfreezes an adapter module.
+
+        Args:
+            adapter_name (str): The name of the adapter to freeze/ unfreeze.
+            freeze (bool, optional): Whether to freeze the adapter. Defaults to True.
+        """
+        if adapter_name in self.refts:
+            self.refts[adapter_name].train(not freeze)
```

Comment:
Why do some of the methods of the Base Class use `self. reft` while others use `self.adapter_modules`?